community/kernel/src/main/java/org/neo4j/kernel/recovery/DefaultRecoverySPI.java

```diff
@@ -88,61 +96,64 @@ public void forceEverything()
     }
 
     @Override
-    public Visitor<LogVersionedStoreChannel,IOException> getRecoverer()
+    public LogPosition getPositionToRecoverFrom() throws IOException
     {
-        return logFileRecoverer;
+        return positionToRecoverFrom.apply( logVersionRepository.getCurrentLogVersion() );
     }
 
     @Override
-    public Iterator<LogVersionedStoreChannel> getLogFiles( final long fromVersion ) throws IOException
+    public Visitor<CommittedTransactionRepresentation,Exception> getRecoveryVisitor()
     {
-        return new Iterator<LogVersionedStoreChannel>()
-        {
-            private long currentVersion = fromVersion;
-            private LogVersionedStoreChannel current =
-                    tryOpenForVersion( logFiles, fileSystemAbstraction, currentVersion );
-
-            @Override
-            public boolean hasNext()
-            {
-                return current != null;
-            }
-
-            @Override
-            public LogVersionedStoreChannel next()
-            {
-                if ( current == null )
-                {
-                    throw new NoSuchElementException();
-                }
-
-                LogVersionedStoreChannel tmp = current;
-                current = tryOpenForVersion( logFiles, fileSystemAbstraction, ++currentVersion );
-                return tmp;
-            }
-
-            @Override
-            public void remove()
-            {
-                throw new UnsupportedOperationException();
-            }
-        };
+        // Calling this method means that recovery is required, tell storage engine about it
+        // This method will be called before recovery actually starts and so will ensure that
+        // each store is aware that recovery will be performed. At this point all the stores have
+        // already started btw.
+        // Go and read more at {@link CommonAbstractStore#deleteIdGenerator()}
+        neoStores.deleteIdGenerators();
```

Comment:
I wanted to remove the constraint that the `recoveryRequired()` needed to be called before `getRecoveryVisitor()`. Renaming `getRecoveryVisitor()` to `startRecovery()` feels better, wdyt?